internal/arr/collector/bazarr_test.go

```diff
@@ -0,0 +1,122 @@
+package collector
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/onedr0p/exportarr/internal/arr/config"
+	"github.com/prometheus/client_golang/prometheus/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+const test_fixtures_path = "../test_fixtures/bazarr/"
+const API_KEY = "abcdef0123456789abcdef0123456789"
+
+func newTestBazarrServer(t *testing.T, fn func(http.ResponseWriter, *http.Request)) (*httptest.Server, error) {
+	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fn(w, r)
+		require.NotEmpty(t, r.URL.Path)
+		// turns /api/some/path into some_path
+		endpoint := strings.Replace(strings.Replace(r.URL.Path, "/api/", "", -1), "/", "_", -1)
+		w.WriteHeader(http.StatusOK)
+		// NOTE: this assumes there is a file that matches the some_path
+		json, err := os.ReadFile(test_fixtures_path + endpoint + ".json")
+		require.NoError(t, err)
+		_, err = w.Write(json)
+		require.NoError(t, err)
+	})), nil
+}
```

Comment:
This approach should be pretty easy to adapt to all the providers for a super generic mocking,

stole it from the sab approach and made it a bit more magic 👍

Reply:
Pretty awesome 😄

Reply:
Thanks for doing this! I keep meaning to get back around to writing tests for the other collectors >.<

Reply:
ill push some followups for the other providers if i get time 👍